main.py

```diff
@@ -0,0 +1,127 @@
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+   contributors may be used to endorse or promote products derived from
+   this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+Requires mkdocs-macros-plugin:
+pip3 install mkdocs-macros-plugin
+
+listfiles() function used in:
+print_compute.md
+print_casters.md
+print_generic.md
+print_sensor_mounts.md
+
+Generates in those files a 3D viewable tree structure of model files.
+
+Set env.variables['branch'] to the branch with your updated file structures pushed, ex: "jinja-ninja".
+Set env.variables['org'] to your github organization name, ex: "rudislabs".
+Set env.variables['render_size_limit'] to max size in kB of stl file for 3D viewing, if above set size include a png image in file structure for viewer to use.
+
+Currently using details in local .git to generate branch and org names.
+'''
+import os
+
+class cl:
+    GRN = '\033[92m'
+    WARN = '\033[93m'
+    FAIL = '\033[91m'
+    END = '\033[0m'
+    BOLD = '\033[1m'
+
+# Begining of optional block and can be commented out if setting env.variables manually
+try:
+    from git import Repo                    
+except ImportError:
+    print('\n\n{:s}{:s}ERROR\t -  Unable to import Python module "git", check to see if GitPython is installed.{:s}\n\n'.format(cl.FAIL,cl.BOLD,cl.END))
+    currentGitBranch = 'main'
+    currentGitOrg = 'iRobotEducation'
+    print('\n{:s}WARN\t -  Using github organization in main.py:{:s} {:s}'.format(cl.WARN,cl.END,currentGitOrg))
+    print('{:s}WARN\t -  Using git branch in main.py:{:s} {:s}\n'.format(cl.WARN,cl.END,currentGitBranch))
+else:
+    repo = Repo(os.getcwd())
+    assert not repo.bare
+    currentGitOrg = repo.remotes.origin.url.split('.git')[0].replace('git@github.com:','').replace('https://github.com/','').split('/')[-2]
+    currentGitBranch = repo.active_branch.name
+    print('INFO\t -  {:s}Using github organization in main.py:{:s} {:s}'.format(cl.GRN,cl.END,currentGitOrg))
+    print('INFO\t -  {:s}Using git branch in main.py:{:s} {:s}'.format(cl.GRN,cl.END,currentGitBranch))
```

Comment:
I think the real useful part is now it all switches to your fork/org/user and current branch you are developing on.

Reply:
And it lets you know branch and org info the slightly bold green: 
![image](https://user-images.githubusercontent.com/10233412/189227087-01a3e92c-75c3-4a1f-9380-a2377e039d49.png)